corehq/apps/domain/tests/test_domain_transfer.py

```diff
@@ -181,10 +181,13 @@ def test_permissions_for_deactivation(self):
         self.assertEqual(resp.status_code, 200)
 
     def test_permissions_for_transfer_domain_view(self):
+        import re
+
         # No one logged in
         resp = self.client.get(reverse('transfer_domain_view',
                                        args=[self.domain.name]), follow=True)
-        self.assertEqual(resp.status_code, 404)
+        self.assertEqual(resp.status_code, 200, 'Should redirect to login')
+        self.assertTrue(re.compile('login'), resp.template_name)
```

Comment:
String matching response template_name could be fragile, I would do
`self.assertRedirects(resp, target_url)`, where `target_url` is `a/fandago/login/?next=/a/fandago/settings/project/transfer/` (could be constructed with `reverse(view)`) . That would take care of, both 302 redirect and target URL.